source/state_representation/src/State.cpp

```diff
@@ -11,6 +12,18 @@ State::State(const StateType& type, const std::string& name, const bool& empty)
 State::State(const State& state)
     : type_(state.type_), name_(state.name_), empty_(state.empty_), timestamp_(std::chrono::steady_clock::now()) {}
 
+void State::set_data(const Eigen::VectorXd& data) {
```

Comment:
I overlooked this in my original review of the PR. If a function argument is unused in the definition, you can remove it (not in the declaration though). This line (and the subsequent "not implemented" ones) can just be:
```c++
void State::set_data(const Eigen::VectorXd&) {
```

Not worth a new PR just for this minor style update, but if one of us happens to do anything around State in future then it'll be nice to fix this.

Reply:
So it won't give a warning when compiling?

Reply:
exactly this is how you ensure no warning there